### ISSUE TITLE
migrate Modulefile → metadata.json

### ISSUE THREAD

**Issue by pcfe** [RedHatSatellite/acme-soe#1]
We still have deprecated `Modulefile`s in this repo.
Migrate to `metadata.json`.

**Comment by pcfe** [comment]
test of ec68c1e finished successfully.

## test report 
### Jenkins side
(note to self: [build 241](http://jenkins.internal.pcfe.net:8080/job/SOE-el7/241/) local brach with all version numbers bumped)
```
#####################################################
#                BUILDING PUPPET MODULES            #
#####################################################
~/workspace/SOE-el7/soe/puppet/auto ~/workspace/SOE-el7/soe/puppet
[mNotice: Building /var/lib/jenkins/workspace/SOE-el7/soe/puppet/auto for release[0m
Module built: /var/lib/jenkins/workspace/SOE-el7/soe/puppet/auto/pkg/acme-auto-0.3.1.tar.gz
‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/auto/pkg/acme-auto-0.3.1.tar.gz’ -> ‘/var/www/html/pub/soe-puppet/acme-auto-0.3.1.tar.gz’
removed ‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/auto/pkg/acme-auto-0.3.1.tar.gz’
~/workspace/SOE-el7/soe/puppet
~/workspace/SOE-el7/soe/puppet/bats ~/workspace/SOE-el7/soe/puppet
[mNotice: Building /var/lib/jenkins/workspace/SOE-el7/soe/puppet/bats for release[0m
Module built: /var/lib/jenkins/workspace/SOE-el7/soe/puppet/bats/pkg/acme-bats-0.0.3.tar.gz
‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/bats/pkg/acme-bats-0.0.3.tar.gz’ -> ‘/var/www/html/pub/soe-puppet/acme-bats-0.0.3.tar.gz’
removed ‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/bats/pkg/acme-bats-0.0.3.tar.gz’
~/workspace/SOE-el7/soe/puppet
~/workspace/SOE-el7/soe/puppet/firewall ~/workspace/SOE-el7/soe/puppet
[mNotice: Building /var/lib/jenkins/workspace/SOE-el7/soe/puppet/firewall for release[0m
Module built: /var/lib/jenkins/workspace/SOE-el7/soe/puppet/firewall/pkg/acme-firewall-0.0.3.tar.gz
‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/firewall/pkg/acme-firewall-0.0.3.tar.gz’ -> ‘/var/www/html/pub/soe-puppet/acme-firewall-0.0.3.tar.gz’
removed ‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/firewall/pkg/acme-firewall-0.0.3.tar.gz’
~/workspace/SOE-el7/soe/puppet
~/workspace/SOE-el7/soe/puppet/ntpd ~/workspace/SOE-el7/soe/puppet
[mNotice: Building /var/lib/jenkins/workspace/SOE-el7/soe/puppet/ntpd for release[0m
Module built: /var/lib/jenkins/workspace/SOE-el7/soe/puppet/ntpd/pkg/acme-ntpd-0.0.2.tar.gz
‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/ntpd/pkg/acme-ntpd-0.0.2.tar.gz’ -> ‘/var/www/html/pub/soe-puppet/acme-ntpd-0.0.2.tar.gz’
removed ‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/ntpd/pkg/acme-ntpd-0.0.2.tar.gz’
~/workspace/SOE-el7/soe/puppet
~/workspace/SOE-el7/soe/puppet/profile ~/workspace/SOE-el7/soe/puppet
[mNotice: Building /var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile for release[0m
Module built: /var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile/pkg/acme-profile-0.3.1.tar.gz
‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile/pkg/acme-profile-0.3.1.tar.gz’ -> ‘/var/www/html/pub/soe-puppet/acme-profile-0.3.1.tar.gz’
removed ‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile/pkg/acme-profile-0.3.1.tar.gz’
~/workspace/SOE-el7/soe/puppet
~/workspace/SOE-el7/soe/puppet/profile_apache ~/workspace/SOE-el7/soe/puppet
[mNotice: Building /var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_apache for release[0m
Module built: /var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_apache/pkg/acme-profile_apache-0.3.1.tar.gz
‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_apache/pkg/acme-profile_apache-0.3.1.tar.gz’ -> ‘/var/www/html/pub/soe-puppet/acme-profile_apache-0.3.1.tar.gz’
removed ‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_apache/pkg/acme-profile_apache-0.3.1.tar.gz’
~/workspace/SOE-el7/soe/puppet
~/workspace/SOE-el7/soe/puppet/profile_base ~/workspace/SOE-el7/soe/puppet
[mNotice: Building /var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_base for release[0m
Module built: /var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_base/pkg/acme-profile_base-0.2.5.tar.gz
‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_base/pkg/acme-profile_base-0.2.5.tar.gz’ -> ‘/var/www/html/pub/soe-puppet/acme-profile_base-0.2.5.tar.gz’
removed ‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_base/pkg/acme-profile_base-0.2.5.tar.gz’
~/workspace/SOE-el7/soe/puppet
~/workspace/SOE-el7/soe/puppet/profile_dmz ~/workspace/SOE-el7/soe/puppet
[mNotice: Building /var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_dmz for release[0m
Module built: /var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_dmz/pkg/acme-dmz-0.0.3.tar.gz
‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_dmz/pkg/acme-dmz-0.0.3.tar.gz’ -> ‘/var/www/html/pub/soe-puppet/acme-dmz-0.0.3.tar.gz’
removed ‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_dmz/pkg/acme-dmz-0.0.3.tar.gz’
~/workspace/SOE-el7/soe/puppet
~/workspace/SOE-el7/soe/puppet/profile_nfs ~/workspace/SOE-el7/soe/puppet
[mNotice: Building /var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_nfs for release[0m
Module built: /var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_nfs/pkg/acme-profile_nfs-0.3.1.tar.gz
‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_nfs/pkg/acme-profile_nfs-0.3.1.tar.gz’ -> ‘/var/www/html/pub/soe-puppet/acme-profile_nfs-0.3.1.tar.gz’
removed ‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_nfs/pkg/acme-profile_nfs-0.3.1.tar.gz’
~/workspace/SOE-el7/soe/puppet
~/workspace/SOE-el7/soe/puppet/profile_postgres ~/workspace/SOE-el7/soe/puppet
[mNotice: Building /var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_postgres for release[0m
Module built: /var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_postgres/pkg/acme-profile_postgres-0.3.1.tar.gz
‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_postgres/pkg/acme-profile_postgres-0.3.1.tar.gz’ -> ‘/var/www/html/pub/soe-puppet/acme-profile_postgres-0.3.1.tar.gz’
removed ‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/profile_postgres/pkg/acme-profile_postgres-0.3.1.tar.gz’
~/workspace/SOE-el7/soe/puppet
~/workspace/SOE-el7/soe/puppet/qpid ~/workspace/SOE-el7/soe/puppet
[mNotice: Building /var/lib/jenkins/workspace/SOE-el7/soe/puppet/qpid for release[0m
Module built: /var/lib/jenkins/workspace/SOE-el7/soe/puppet/qpid/pkg/acme-qpid-0.0.2.tar.gz
‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/qpid/pkg/acme-qpid-0.0.2.tar.gz’ -> ‘/var/www/html/pub/soe-puppet/acme-qpid-0.0.2.tar.gz’
removed ‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/qpid/pkg/acme-qpid-0.0.2.tar.gz’
~/workspace/SOE-el7/soe/puppet
~/workspace/SOE-el7/soe/puppet/role ~/workspace/SOE-el7/soe/puppet
[mNotice: Building /var/lib/jenkins/workspace/SOE-el7/soe/puppet/role for release[0m
Module built: /var/lib/jenkins/workspace/SOE-el7/soe/puppet/role/pkg/acme-role-0.2.1.tar.gz
‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/role/pkg/acme-role-0.2.1.tar.gz’ -> ‘/var/www/html/pub/soe-puppet/acme-role-0.2.1.tar.gz’
removed ‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/role/pkg/acme-role-0.2.1.tar.gz’
~/workspace/SOE-el7/soe/puppet
~/workspace/SOE-el7/soe/puppet/role_db ~/workspace/SOE-el7/soe/puppet
[mNotice: Building /var/lib/jenkins/workspace/SOE-el7/soe/puppet/role_db for release[0m
Module built: /var/lib/jenkins/workspace/SOE-el7/soe/puppet/role_db/pkg/acme-role_db-0.3.1.tar.gz
‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/role_db/pkg/acme-role_db-0.3.1.tar.gz’ -> ‘/var/www/html/pub/soe-puppet/acme-role_db-0.3.1.tar.gz’
removed ‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/role_db/pkg/acme-role_db-0.3.1.tar.gz’
~/workspace/SOE-el7/soe/puppet
~/workspace/SOE-el7/soe/puppet/role_www ~/workspace/SOE-el7/soe/puppet
[mNotice: Building /var/lib/jenkins/workspace/SOE-el7/soe/puppet/role_www for release[0m
Module built: /var/lib/jenkins/workspace/SOE-el7/soe/puppet/role_www/pkg/acme-role_www-0.2.1.tar.gz
‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/role_www/pkg/acme-role_www-0.2.1.tar.gz’ -> ‘/var/www/html/pub/soe-puppet/acme-role_www-0.2.1.tar.gz’
removed ‘/var/lib/jenkins/workspace/SOE-el7/soe/puppet/role_www/pkg/acme-role_www-0.2.1.tar.gz’
~/workspace/SOE-el7/soe/puppet
#####################################################
#                BUILDING KICKSTARTS                #
#####################################################
[SOE-el7] $ /bin/bash /tmp/jenkins943485332055694868.sh
#####################################################
#                PUSH RPMS TO SATELLITE             #
#####################################################
INFO:    No entries in /var/lib/jenkins/workspace/SOE-el7/modified_rpms.track no need to continue with /var/lib/jenkins/workspace/SOE-el7/scripts/rpmpush.sh [Sat Oct 21 14:34:31 CEST 2017]
#####################################################
#         PUSH PUPPET MODULES TO SATELLITE          #
#####################################################
Task f019001d-1c7e-46ab-ac08-b2a7c704d823 planned: 0.0/1, 0%, elapsed: 00:00:00
Task f019001d-1c7e-46ab-ac08-b2a7c704d823 running: 0.0/1, 0%, 0.0/s, elapsed: 00:00:02
Task f019001d-1c7e-46ab-ac08-b2a7c704d823 running: 0.5/1, 50%, 0.1/s, elapsed: 00:00:04, ETA: 00:00:04
Task f019001d-1c7e-46ab-ac08-b2a7c704d823 running: 0.5/1, 50%, 0.1/s, elapsed: 00:00:06, ETA: 00:00:06
Task f019001d-1c7e-46ab-ac08-b2a7c704d823 running: 0.6/1, 60%, 0.1/s, elapsed: 00:00:08, ETA: 00:00:05
Task f019001d-1c7e-46ab-ac08-b2a7c704d823 running: 0.65/1, 65%, 0.1/s, elapsed: 00:00:10, ETA: 00:00:05
Task f019001d-1c7e-46ab-ac08-b2a7c704d823 success: 1.0/1, 100%, 0.1/s, elapsed: 00:00:12
Task f019001d-1c7e-46ab-ac08-b2a7c704d823 success: 1.0/1, 100%, 0.1/s, elapsed: 00:00:12
Total module count: 14.
```

### Satellite 6.2.12 side
Modules on Sat 6.2 after that build (all version numbers were bumped to force a rebuild)
```
[jenkins@satellite ~]$ hammer puppet-module list
---|------------------|------------|---------|-------------------------------------
ID | NAME             | AUTHOR     | VERSION | UUID                                
---|------------------|------------|---------|-------------------------------------
34 | profile_postgres | acme       | 0.3.1   | fc500e7e-d258-4b04-84de-1822d5ffd4ab
33 | bats             | acme       | 0.0.3   | f5dbe522-f2e0-4669-a42b-33f833696913
32 | auto             | acme       | 0.3.1   | dc732cfa-8b4b-4b21-84f1-7aa96b61cc63
31 | profile_apache   | acme       | 0.3.1   | aeda3ab1-2b77-4c39-9bde-720e7787150a
30 | dmz              | acme       | 0.0.3   | a30dd522-dbf2-4df8-8871-92d34b0d317b
29 | role_db          | acme       | 0.3.1   | 96e70e5d-9839-4610-a7d2-ee4e28ab5521
28 | firewall         | acme       | 0.0.3   | 76cc8107-ef61-4893-abf2-842fe04e8057
27 | profile          | acme       | 0.3.1   | 4144c22f-864a-4184-85eb-e532cfdca8b5
26 | role_www         | acme       | 0.2.1   | 3a32339b-a276-45a9-8129-ba3110ba2ffb
25 | role             | acme       | 0.2.1   | 38e53786-24fd-41ee-98bc-adfdf6c20817
24 | qpid             | acme       | 0.0.2   | 23a48990-a573-43fe-9af4-47821656f5a1
23 | ntpd             | acme       | 0.0.2   | 229c279f-6f93-4c90-a0b8-4400cf656f08
22 | profile_base     | acme       | 0.2.5   | 0e60a1c1-a333-4b81-a615-63e59f083062
21 | profile_nfs      | acme       | 0.3.1   | 0865f365-1579-4b98-8879-0fd97815af9f
20 | firewall         | puppetlabs | 1.9.0   | 19e6c4aa-9702-43bf-8167-f950d487576e
19 | common           | ghoneycutt | 1.7.0   | ea96c69f-539d-49d6-ad3d-a9cf41c99193
18 | ssh              | ghoneycutt | 3.54.0  | 4ad79cc1-4f6a-4b3d-b781-8578605de4a2
16 | bash             | kemra102   | 2.0.0   | 754c6250-badc-4e4d-ba8c-c9e9e98f2582
15 | ntpd             | acme       | 0.0.1   | 305cf5e0-0017-4c30-8874-1ef55247d00a
14 | profile_postgres | acme       | 0.3.0   | e5a980b1-aa3b-4140-b21b-679e7aa13429
13 | firewall         | acme       | 0.0.2   | c6b855f7-fbe1-412d-9fd2-0d03105ae2c8
12 | profile_base     | acme       | 0.2.4   | c2d900b5-26c8-433d-91d0-395af7986633
11 | profile_apache   | acme       | 0.3.0   | b97f6c21-7a4f-4f7d-965e-9199ebba0ddd
10 | role_db          | acme       | 0.3.0   | 7c986754-f4ca-4304-a889-e7a22773fa8a
9  | role_www         | acme       | 0.2.0   | 7669e6d9-1f83-4b9c-bc1b-9be9b52073c7
8  | auto             | acme       | 0.3.0   | 73f860ec-9438-49bf-a8aa-7a1fb9e8d830
6  | profile          | acme       | 0.3.0   | 49706380-490c-480b-b383-732a6bde169f
5  | profile_nfs      | acme       | 0.3.0   | 2c85fb94-e598-40ac-b336-b9d060f9db96
4  | bats             | acme       | 0.0.2   | 2af478d5-6535-4cba-a5b0-b095526fa5ff
3  | qpid             | acme       | 0.0.1   | 24f4a7f6-b160-48de-be95-8f615275f508
2  | role             | acme       | 0.2.0   | 17464641-5a12-4897-8635-cd9bb135fe88
1  | dmz              | acme       | 0.0.2   | 0db567de-0f17-4bea-ba71-1f24ef1091b9
---|------------------|------------|---------|-------------------------------------
```

**Comment by pcfe** [comment]
fixed by 5a4eb17e6438d20775a13a0f500f19431a4aaffe
